worker/src/router.rs

```diff
@@ -1,403 +0,0 @@
-use std::{collections::HashMap, future::Future, rc::Rc};
-
-use futures_util::future::LocalBoxFuture;
-use matchit::{Match, Node};
-use worker_kv::KvStore;
-
-use crate::{
-    durable::ObjectNamespace,
-    env::{Env, Secret, Var},
-    http::Method,
-    request::Request,
-    response::Response,
-    Bucket, Fetcher, Result,
-};
-
-type HandlerFn<D> = fn(Request, RouteContext<D>) -> Result<Response>;
-type AsyncHandlerFn<'a, D> =
-    Rc<dyn 'a + Fn(Request, RouteContext<D>) -> LocalBoxFuture<'a, Result<Response>>>;
-
-/// Represents the URL parameters parsed from the path, e.g. a route with "/user/:id" pattern would
-/// contain a single "id" key.
-pub struct RouteParams(HashMap<String, String>);
-
-impl RouteParams {
-    fn get(&self, key: &str) -> Option<&String> {
-        self.0.get(key)
-    }
-}
-
-enum Handler<'a, D> {
-    Async(AsyncHandlerFn<'a, D>),
-    Sync(HandlerFn<D>),
-}
-
-impl<D> Clone for Handler<'_, D> {
-    fn clone(&self) -> Self {
-        match self {
-            Self::Async(rc) => Self::Async(rc.clone()),
-            Self::Sync(func) => Self::Sync(*func),
-        }
-    }
-}
-
-/// A path-based HTTP router supporting exact-match or wildcard placeholders and shared data.
-pub struct Router<'a, D> {
```

Comment:
Idk how I feel about removing this, I think it's less useful now that we can use axum but many people might not want to use axum. I think gating this behind a `router` feature flag would be better, at least until there's another routing crate with similar behavior.

Reply:
I personally don't think this library should provide a router type. Perhaps publishing it as a separate crate would be better?